Sofa/Component/Constraint/Lagrangian/Solver/src/sofa/component/constraint/lagrangian/solver/GenericConstraintSolver.cpp

```diff
@@ -69,13 +69,14 @@ void clearMultiVecId(sofa::core::objectmodel::BaseContext* ctx, const sofa::core
 }
 
 GenericConstraintSolver::GenericConstraintSolver()
-    : maxIt( initData(&maxIt, 1000, "maxIterations", "maximal number of iterations of the Gauss-Seidel algorithm"))
+    : d_resolutionMethod( initData(&d_resolutionMethod, "resolutionMethod", "Method used to solve the Non-Linear Complementary Problem (NLCP) among: \"ProjectedGaussSeidel\" or \"for NonsmoothNonlinearConjugateGradient\""))
+    , maxIt( initData(&maxIt, 1000, "maxIterations", "maximal number of iterations of the Gauss-Seidel algorithm"))
     , tolerance( initData(&tolerance, 0.001, "tolerance", "residual error threshold for termination of the Gauss-Seidel algorithm"))
     , sor( initData(&sor, 1.0, "sor", "Successive Over Relaxation parameter (0-2)"))
     , scaleTolerance( initData(&scaleTolerance, true, "scaleTolerance", "Scale the error tolerance with the number of constraints"))
     , allVerified( initData(&allVerified, false, "allVerified", "All contraints must be verified (each constraint's error < tolerance)"))
-    , schemeCorrection( initData(&schemeCorrection, false, "schemeCorrection", "Apply new scheme where compliance is progressively corrected"))
-    , unbuilt(initData(&unbuilt, false, "unbuilt", "Compliance is not fully built"))
+    , unbuilt(initData(&unbuilt, false, "unbuilt", "Compliance is not fully built  (for the ProjectedGaussSeidel solver only)"))
```

Comment:
It seems logical (to me) that the unbuilt method should be a third option in `d_resolutionMethod`.

If you look at line 271: if you are using `NonsmoothNonlinearConjugateGradient`, you still get two different branches based on the `unbuilt` condition. `unbuilt` should not influence `NonsmoothNonlinearConjugateGradient` if I understood correctly.